### PR TITLE
Merge 34 n to test

### PR DESCRIPTION
**Pull request by japshvincent**

### CODE REVIEW

**Comment by japshvincent** [comment]
https://jenkins.ome.ksu.edu/job/Java/job/Test%20Deployers/job/LTI%20K-State%20Attendance%20(test1)/41/console

File: src/main/resources/static/css/buttonOverrides.css [hunk 1]
```diff
@@ -1,3 +1,25 @@
+.button_disabled
```
Comment (KSU-Github) [inline src/main/resources/static/css/buttonOverrides.css]: MINOR Rename selector "button_disabled" to match the regular expression: ^[a-z][-a-z0-9]*$ [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=css%3Aselector-naming-convention)

File: src/main/resources/static/css/buttonOverrides.css [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+{
```
Comment (KSU-Github) [inline src/main/resources/static/css/buttonOverrides.css]: MINOR Move the opening curly brace to the previous line. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=css%3Aformatting)

File: src/main/resources/static/css/buttonOverrides.css [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+    -webkit-user-select: none;
+
+    color: #fff;
+    background-color: #D3D3D3;
+    border-color: #999999;
+
+}
+
 .hovering-purple-button
 {
```
Comment (KSU-Github) [inline src/main/resources/static/css/buttonOverrides.css]: MINOR Move the opening curly brace to the previous line. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=css%3Aformatting)

File: src/main/resources/static/css/buttonOverrides.css [hunk 4]
```diff
@@ -1,3 +1,25 @@
...
+    text-align: center;
+    vertical-align: middle;
+    cursor: pointer;
+    background-image: none;
+    border: 1px solid transparent;
+    padding: 6px 12px;
+    font-size: 14px;
+    line-height: 1.4;
+    border-radius: 4px;
+    -webkit-user-select: none;
```
Comment (KSU-Github) [inline src/main/resources/static/css/buttonOverrides.css]: MAJOR Define the standard property right after this vendor-prefixed property. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=css%3Avendor-prefix)
MAJOR Remove this usage of the experimental "user-select" property. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=css%3Aexperimental-property-usage)

**Comment by KSU-Github** [comment]
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 10 minor

Watch the comments in this conversation to review them.

#### 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [buttonOverrides.css#L37](https://github.com/kstateome/lti-attendance/blob/0345980678fbf3878d15b306ce5ee12eaa4b8e26/src/main/resources/static/css/buttonOverrides.css#L37): Define the standard property right after this vendor-prefixed property. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=css%3Avendor-prefix)
1. ![MAJOR][MAJOR] [buttonOverrides.css#L37](https://github.com/kstateome/lti-attendance/blob/0345980678fbf3878d15b306ce5ee12eaa4b8e26/src/main/resources/static/css/buttonOverrides.css#L37): Remove this usage of the experimental "user-select" property. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=css%3Aexperimental-property-usage)
1. ![MAJOR][MAJOR] [courseConfiguration.jsp#L8](https://github.com/kstateome/lti-attendance/blob/0345980678fbf3878d15b306ce5ee12eaa4b8e26/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L8): Add a 'favicon' declaration in this 'header' tag. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3APageWithoutFaviconCheck)
1. ![MINOR][MINOR] [AttendanceSectionService.java#L55](https://github.com/kstateome/lti-attendance/blob/0345980678fbf3878d15b306ce5ee12eaa4b8e26/src/main/java/edu/ksu/canvas/attendance/services/AttendanceSectionService.java#L55): Define a constant instead of duplicating this literal "Cannot load data into courseForm for non-existent sections for this course" 3 times. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [AttendanceSectionService.java#L56](https://github.com/kstateome/lti-attendance/blob/0345980678fbf3878d15b306ce5ee12eaa4b8e26/src/main/java/edu/ksu/canvas/attendance/services/AttendanceSectionService.java#L56): Define a constant instead of duplicating this literal "courseId" 3 times. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [buttonOverrides.css#L46](https://github.com/kstateome/lti-attendance/blob/0345980678fbf3878d15b306ce5ee12eaa4b8e26/src/main/resources/static/css/buttonOverrides.css#L46): Move the opening curly brace to the previous line. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=css%3Aformatting)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L108](https://github.com/kstateome/lti-attendance/blob/0345980678fbf3878d15b306ce5ee12eaa4b8e26/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L108): Add a <legend> tag to this fieldset. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3AFieldsetWithoutLegendCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L142](https://github.com/kstateome/lti-attendance/blob/0345980678fbf3878d15b306ce5ee12eaa4b8e26/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L142): Replace this <i> tag by <em>. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3ABoldAndItalicTagsCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L147](https://github.com/kstateome/lti-attendance/blob/0345980678fbf3878d15b306ce5ee12eaa4b8e26/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L147): Replace this <i> tag by <em>. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3ABoldAndItalicTagsCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L153](https://github.com/kstateome/lti-attendance/blob/0345980678fbf3878d15b306ce5ee12eaa4b8e26/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L153): Replace this <i> tag by <em>. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3ABoldAndItalicTagsCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**